components/brave_new_tab_ui/components/default/cryptoDotCom/index.tsx

```diff
@@ -0,0 +1,459 @@
+              <FlexItem $pl={5} $pr={5}>
+                {this.renderIconAsset(currency.toLowerCase())}
+              </FlexItem>
+              <FlexItem>
+                <Text>{currency}</Text>
+                <Text small={true} textColor='light'>{currencyNames[currency]}</Text>
+              </FlexItem>
+              <FlexItem textAlign='right' flex={1}>
+                {(price !== null) && <Text>{this.formattedNum(price)}</Text>}
+                {(percentChange !== null) && <Text textColor={percentChange > 0 ? 'green' : 'red'}>{percentChange}%</Text>}
+              </FlexItem>
+            </ListItem>
+          )
+        })}
+      </List>
+    )
+  }
+
+  renderAssetDetailView () {
+    const { selectedAsset: currency } = this.state
+    const { price = null, volume = null } = this.props.tickerPrices[currency] || {}
+    const chartData = this.props.charts[currency] || []
+    const pairs = this.props.supportedPairs[currency] || []
+
+    const losersGainers = this.transformLosersGainers(this.props.losersGainers || {})
+    const { percentChange = null } = losersGainers[currency] || {}
+
+    const chartHeight = 100
+    const chartWidth = 309
+    return (
+      <Box hasPadding={false}>
+        <FlexItem
+          hasPadding={true}
+          isFlex={true}
+          isFullWidth={true}
+          hasBorder={true}
+        >
+          <FlexItem>
+            <BackArrow>
+              <CaratLeftIcon onClick={this.setSelectedAsset.bind(this, '')} />
+            </BackArrow>
+          </FlexItem>
+          <FlexItem $pr={5}>
+            {this.renderIconAsset(currency.toLowerCase())}
+          </FlexItem>
+          <FlexItem flex={1}>
+            <Text>{currency}</Text>
+            <Text small={true} textColor='light'>
+              {currencyNames[currency]}
+            </Text>
+          </FlexItem>
+          <FlexItem $pl={5}>
+            <ActionButton onClick={this.onClickBuyPair.bind(this, `${currency}_USDT`)} small={true} light={true}>
+              <UpperCaseText>
+                {getLocale('cryptoDotComWidgetBuy')}
+              </UpperCaseText>
+            </ActionButton>
+          </FlexItem>
+        </FlexItem>
+        <FlexItem
+          hasPadding={true}
+          isFullWidth={true}
+          hasBorder={true}
+        >
+          {(price !== null) && <Text
+            inline={true}
+            large={true}
+            weight={500}
+            $mr='0.5rem'
+          >
+            {this.formattedNum(price)} USDT
+          </Text>}
+          {(percentChange !== null) && <Text inline={true} textColor={percentChange > 0 ? 'green' : 'red'}>{percentChange}%</Text>}
+          <SVG viewBox={`0 0 ${chartWidth} ${chartHeight}`}>
+            <polyline
+              fill='none'
+              stroke='#44B0FF'
+              strokeWidth='3'
+              points={this.plotData({
```

Comment:
Is it ok to calculate `plotData` on every single render instead of only when its dependent data changes? Are there any props in this component which will cause the component to be re-rendered rapidly and these calculations to be done more than necessary?

Reply:
On testing, it doesn't re-render excessively (even less after refactoring the redux actions as per your previous comments). We could memoize the function, but I don't think it's heavy enough to warrant the added complexity.